### PR TITLE
feat: Add regular expression host analyser

### PR DESCRIPTION
**Pull request by banjoh**
This analyser is exactly the same as the in-cluster analyser

Related to https://github.com/replicatedhq/troubleshoot/issues/1263

In addition to the new analyser change, I reorganised the TOC to make it have an alphabetical order

### CODE REVIEW

File: docs/gatsby-config.js [hunk 1]
```diff
@@ -42,75 +42,74 @@ const themeOptions = {
...
+              'analyze/node-resources',
+              'analyze/postgresql',
+              'analyze/regex',
+              'analyze/replicaset-status',
+              'analyze/storage-class',
+              'analyze/configmap',
+              'analyze/yaml-compare',
               'analyze/registry-images',
               'analyze/weave-report',
+              'analyze/statefulset-status',
```
Comment (xavpaice) [inline docs/gatsby-config.js:105]: there's a lot of re-arrangement here, is this intended?

**Review by paigecalvert — DISMISSED** [review]
Made a couple minor edits and added a suggestion for breaking up that topic sentence a bit. looks good!

File: docs/gatsby-config.js (hunk 1 above)
Comment (banjoh) [reply docs/gatsby-config.js:105]: > there's a lot of re-arrangement here, is this intended?

Yes. I rearranged the TOC to have it in alphabetical order.

File: docs/source/host-collect-analyze/regex.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+title: Regular Expression
+description: Using a regular expression to analyze arbitrary data
+---
+
+The regex analyzer is used to run arbitrary regular expressions against data collected into a bundle by any collector that collects text data such as static data, run, copy, pod logs or exec collectors.
```
[the remark this answers is not in the document]
Comment (xavpaice) [reply docs/source/host-collect-analyze/regex.md:6]: I see the docs/source/analyze/regex.md sentence got an edit, but not this one

File: docs/gatsby-config.js (hunk 1 above)
Comment (xavpaice) [reply docs/gatsby-config.js:105]: if that's a pattern used throughout, makes sense

**Review by xavpaice — CHANGES_REQUESTED** [review]

**Comment by banjoh** [comment]
@xavpaice requested change made

**Review by xavpaice — APPROVED** [review]

**Pull request merged**